.clang-tidy-fixes

```diff
@@ -0,0 +1,87 @@
+# This file contains only checks that offer fixes. It is intended to be run with the '-fix' option.
```

Comment:
This is the kind of comment that gets out of date very fast 😄 
If we go for a hard rule, as the one here, we'll have to maintain the list for each new clang-tidy version—any checks that didn't offer fixes but do now? Any new checks, and if so, do they offer fixes?
The old file might be a useful tradeoff; by enabling and disabling whole categories it lets checks slip in even if they don't provide fixes, but it needs less maintaining.  What do you think?

Also, I would keep this file as `.clang-tidy` (in case some IDEs pick it up by default) and give the stricter one a different name, so it's an opt-in as we discussed.

Reply:
I agree that it's going to be messy to continually maintain the list of checks that offer fixes. I thought that one of the goals of splitting the fixes from the non-fixes was to reduce the computing time spent on non-fixes, but if that's not a big concern then indeed let's include all of the checks for each category by default and only exclude the ones that don't apply to the project. In my view, it probably shouldn't be a big concern now that the clang-tidy job runs on a nightly schedule, instead of for every pull request.

By the way, I created https://github.com/llvm/llvm-project/issues/54939 to make it easier for us to only run the checks that offer fixes, and if this ever gets picked up then we can try to use it for the nightly job.

Regarding the default .clang-tidy file, my thinking was that anybody who runs clang-tidy in an IDE either through the default integration or through the `QL_USE_CLANG_TIDY` CMake option would be interested in all of the checks by default, not just the ones that offer fixes. Anyone who is not interested in clang-tidy can just simply not run it as part of their local build. So in that sense the additional checks are still 'opt-in' because the default build configuration (at least for all the IDEs that I know about) is not to run clang-tidy. But if you still prefer, I will changethe file names.

Reply:
You're right, I was still thinking of the old context.  Now that's nightly (it could be weekly) we can have the one file with all the checks.